modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/action/ContentDashboardItemActionTrackerImpl.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.web.internal.item.action;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemAction;
+import com.liferay.content.dashboard.item.action.ContentDashboardItemActionProvider;
+import com.liferay.content.dashboard.item.action.ContentDashboardItemActionTracker;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+/**
+ * @author David Arques
+ */
+@Component(immediate = true, service = ContentDashboardItemActionTracker.class)
+public class ContentDashboardItemActionTrackerImpl
+	implements ContentDashboardItemActionTracker {
+
+	@Override
+	public List<ContentDashboardItemAction> getContentDashboardItemActions(
+		String className, long classPK, HttpServletRequest httpServletRequest,
+		Locale locale) {
+
+		Collection<ContentDashboardItemActionProvider>
```

Comment:
Should we track the providers? won't be easier to track the action interface itself?